pennylane/grouping/transformations.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Functions for finding the unitary transformations required to diagonalize Pauli partitions in the
+measurement basis, and templates for the circuit implementations.
+"""
+import pennylane as qml
+from pennylane.operation import Tensor
+from pennylane.wires import Wires
+from pennylane.templates import template
+from pennylane.grouping.utils import pauli_to_binary, is_qwc
+import numpy as np
+
+
+@template
+def qwc_rotation(pauli_dict):
+    """Performs circuit implementation of diagonalizing unitary for a Pauli string.
+
+    **Usage example:**
+
+    >>> pauli_dict = {0: qml.PauliX(0), 2: qml.PauliY(2), 3: qml.PauliZ(3)}
+    >>> qwc_rotation(pauli_dict)
+    [RY(-1.5707963267948966, wires=[0]), RX(1.5707963267948966, wires=[2])]
+
+    Args:
+        pauli_dict (dict[int, Observable]): key is the wire (int), and value is a qml.PauliX,
+            qml.PauliY, or qml.PauliZ instance.
```

Comment:
Believe that `Wire` instance would have to be accepted here, correct? :thinking: 

Found the description a bit difficult to grasp at first, perhaps could be rephrased? E.g. `mapping between Wire instances and qml.PauliX, qml.PauliY, qml.PauliZ, qml.Identity instances` or comparable

```suggestion
        pauli_dict (dict[int, Observable]): mapping between Wire instances and qml.PauliX,
            qml.PauliY, qml.PauliZ, qml.Identity instances.
```
```suggestion
        pauli_dict (dict[Wire, Observable]): key is the wire (int), and value is a qml.PauliX,
            qml.PauliY, or qml.PauliZ instance.
```

Reply:
Yep, you're totally right. The docstring was outdated. Just to confirm, it should be `Wires` as opposed to `Wire`, correct?

Reply:
Correct, good catch!